File: app/src/main/java/com/example/administrator/encyclopediamvp/View/LucklyQQ/LucklyQQActivity.java
package com.example.administrator.encyclopediamvp.View.LucklyQQ;

import android.content.Intent;
import android.os.Looper;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import com.example.administrator.encyclopediamvp.Bean.LuckyQQBean;
import com.example.administrator.encyclopediamvp.Presenter.LucklyQQPresenter;
import com.example.administrator.encyclopediamvp.R;
import com.example.administrator.encyclopediamvp.View.IBaseView;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.BindViews;
import butterknife.ButterKnife;

public class LucklyQQActivity extends AppCompatActivity implements ILucklyQQView , OnClickListener{

    //绑定控件
    @BindView(R.id.serch_button)Button mSerch;
    @BindView(R.id.serch_input)EditText mEditQQ;
    @BindView(R.id.toolbar)Toolbar toolbar;
    @BindView(R.id.toolbar_title)TextView mTitle;
    @BindView(R.id.toolbar_back)Button mBackBtn;
    @BindViews({R.id.Luck_QQNum,R.id.Luck_QQScore,R.id.Luck_QQgrade,R.id.Luck_QQdesc,R.id.Luck_QQanalysis})List<TextView> mResult;

    //数据
    LucklyQQPresenter presenter;
    LuckyQQBean Luckbean = new LuckyQQBean();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_luckly_qq);
        ButterKnife.bind(this);
        presenter = new LucklyQQPresenter(this);
        Intent i = getIntent();
        mTitle.setText(i.getStringExtra("title"));
        mSerch.setOnClickListener(this);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayShowTitleEnabled(false);
        mBackBtn.setOnClickListener(this);



    }

    @Override
    public void onClick(View v) {
        switch (v.getId()){
            case R.id.toolbar_back:
                finish();
                break;
            case R.id.serch_button:

                String QQ = mEditQQ.getText().toString().trim();
                presenter.SetResult(QQ);
                break;
        }
    }

    @Override
    public void setData(LuckyQQBean bean) {

        Log.i("JSY", "V:"+bean.getShowapi_res_body().getDesc());

        if(bean != null){
            Luckbean = bean;
            if(bean.getShowapi_res_body() !=null){
                Luckbean.setShowapi_res_body(bean.getShowapi_res_body());
                 mResult.get(0).setText("QQ:"+mEditQQ.getText().toString().trim());
                 mResult.get(1).setText("评分:"+Luckbean.getShowapi_res_body().getScore());
                 mResult.get(2).setText("评价:"+Luckbean.getShowapi_res_body().getGrade());
                 mResult.get(3).setText("分析:"+Luckbean.getShowapi_res_body().getDesc());
                 mResult.get(4).setText("建议:"+Luckbean.getShowapi_res_body().getAnalysis());
                if(bean.getShowapi_res_body().getScore() != null){

                }else{
                    Log.i("JSY", "V层Bean中Body的值为null");
                }

            }else{
                Log.i("JSY", "V层Bean中Body返回体为null");
            }
        }else{
            Log.i("JSY", "V层Bean为null");
        }




    }
}

File: app/src/main/java/com/example/administrator/encyclopediamvp/View/LucklyQQ/LucklyQQActivity.kt
package com.example.administrator.encyclopediamvp.View.LucklyQQ

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.view.View.OnClickListener
import com.example.administrator.encyclopediamvp.Bean.LuckyQQBean
import com.example.administrator.encyclopediamvp.Presenter.LucklyQQPresenter
import com.example.administrator.encyclopediamvp.R
import kotlinx.android.synthetic.main.activity_luckly_qq.*
import kotlinx.android.synthetic.main.include_back.*
import kotlinx.android.synthetic.main.layout_serch.*

class LucklyQQActivity : AppCompatActivity(), ILucklyQQView, OnClickListener {

    //数据
    var presenter: LucklyQQPresenter? = null
    var Luckbean = LuckyQQBean()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_luckly_qq)
        setSupportActionBar(toolbar)
        presenter = LucklyQQPresenter(this)
        val i = intent
        toolbar_title.text = i.getStringExtra("title")
        serch_button.setOnClickListener(this)
        supportActionBar?.setDisplayShowTitleEnabled(false)
        toolbar_back.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.toolbar_back -> finish()
            R.id.serch_button -> {
                var QQ = serch_input.text.toString().trim()
                presenter!!.SetResult(QQ)
            }
        }
    }

    override fun setData(bean: LuckyQQBean) {
        Luckbean = bean!!
        Luckbean.showapi_res_body = bean.showapi_res_body
        Luck_QQNum.text = "QQ:" + serch_input.text.toString().trim()!!
        Luck_QQScore.text = "评分:" + Luckbean.showapi_res_body!!.score
        Luck_QQgrade.text = "评价:" + Luckbean.showapi_res_body!!.grade
        Luck_QQdesc.text = "分析:" + Luckbean.showapi_res_body!!.desc
        Luck_QQanalysis.text = "建议:" + Luckbean.showapi_res_body!!.analysis


    }
}
